### ISSUE TITLE
MicroProfile Reactive Messaging 2.0 Compatible Certification Request

### ISSUE THREAD

**Issue by cescoffier** [eclipse/microprofile-reactive-messaging#134]
- [X] Organization Name ("Organization") and, if applicable, URL:<br/>
Red Hat

- [X] Product Name, Version and download URL (if applicable):<br/>

* SmallRye Reactive Messaging 3.3.0.RC1
* https://oss.sonatype.org/content/repositories/iosmallrye-1557/

- [X] Specification Name, Version and download URL:<br/>

MicroProfile Reactive Messaging 2.0

- [X] Public URL of TCK Results Summary:<br/>

* https://github.com/smallrye/smallrye-reactive-messaging/actions/runs/798899367
* Output of the TCK run - Java 11:
[tck.txt](https://github.com/eclipse/microprofile-reactive-messaging/files/6411906/tck.txt)

- [X] Java runtime used to run the implementation:<br/>

* Java 8, 11, 15, 16

- [ ] Summary of the information for the certification environment, operating system, cloud, ...:<br/>

Github Action, using Ubuntu-20.04

- [X] By checking this box I acknowledge that the Organization I represent accepts the terms of the [EFTL](https://www.eclipse.org/legal/tck.php).
- [X] By checking this box I attest that all TCK requirements have been met, including any compatibility rules.

**Comment by cescoffier** [comment]
@radcortez Can you double check?

**Comment by radcortez** [comment]
Looks good. A couple of notes:

- Are we using this for a final release of Reactive Messaging 2.0? If so, I believe the WG would not accept the CI to be in the staging repo when validating the binaries (we still need to change this into a more manageable release process).
- If you want to attach a TCK test run file, for do it for Java 8, because that is still official the Java version targeted by the MP platform (other versions are fine of course).

**Comment by cescoffier** [comment]
> Looks good. A couple of notes:
> 
> * Are we using this for a final release of Reactive Messaging 2.0? If so, I believe the WG would not accept the CI to be in the staging repo when validating the binaries (we still need to change this into a more manageable release process).

No, it's using an RC. Even if the RC is in Maven Central, I really don't want to have this release be in Maven Central. It's there to validate that we pass the TCK, not something to be used by the end-user. There are many more steps to do before that point. With dependabot, if we push this into Central, you can be sure that tomorrow, you get people updating. 

> * If you want to attach a TCK test run file, for do it for Java 8, because that is still officially the Java version targeted by the MP platform (other versions are fine of course).

Ok, note that I plan to drop Java 8, soon, really soon.

**Comment by cescoffier** [comment]
I have added the Java 8 output.

**Comment by radcortez** [comment]
> > Looks good. A couple of notes:
> > 
> > * Are we using this for a final release of Reactive Messaging 2.0? If so, I believe the WG would not accept the CI to be in the staging repo when validating the binaries (we still need to change this into a more manageable release process).
> 
> No, it's using an RC. Even if the RC is in Maven Central, I really don't want to have this release be in Maven Central. It's there to validate that we pass the TCK, not something to be used by the end-user. There are many more steps to do before that point. With dependabot, if we push this into Central, you can be sure that tomorrow, you get people updating.
> 

I agree. I don't like us requiring it as well, but let's see if we can do it this way.

**Comment by Emily-Jiang** [comment]
> > > Looks good. A couple of notes:
> > > 
> > > * Are we using this for a final release of Reactive Messaging 2.0? If so, I believe the WG would not accept the CI to be in the staging repo when validating the binaries (we still need to change this into a more manageable release process).
> > 
> > 
> > No, it's using an RC. Even if the RC is in Maven Central, I really don't want to have this release be in Maven Central. It's there to validate that we pass the TCK, not something to be used by the end-user. There are many more steps to do before that point. With dependabot, if we push this into Central, you can be sure that tomorrow, you get people updating.
> 
> I agree. I don't like us requiring it as well, but let's see if we can do it this way.

This was discussed on [the mailinglist](https://groups.google.com/g/microprofile/c/6v65mBjIXiQ) and we settled on RCx reelase for CCR as long as the final release is same as RCx. 

This CCR looks good. I will kick off the release process with this.

**Comment by radcortez** [comment]
> > > > Looks good. A couple of notes:
> > > > 
> > > > * Are we using this for a final release of Reactive Messaging 2.0? If so, I believe the WG would not accept the CI to be in the staging repo when validating the binaries (we still need to change this into a more manageable release process).
> > > 
> > > 
> > > No, it's using an RC. Even if the RC is in Maven Central, I really don't want to have this release be in Maven Central. It's there to validate that we pass the TCK, not something to be used by the end-user. There are many more steps to do before that point. With dependabot, if we push this into Central, you can be sure that tomorrow, you get people updating.
> > 
> > 
> > I agree. I don't like us requiring it as well, but let's see if we can do it this way.
> 
> This was discussed on [the mailinglist](https://groups.google.com/g/microprofile/c/6v65mBjIXiQ) and we settled on RCx reelase for CCR as long as the final release is same as RCx.
> 
> This CCR looks good. I will kick off the release process with this.

I'm not completely sure if everyone has the same understanding. I'm fine like this, but we may need to document this and put it through an approval to make sure all of use are on the same page.

**Comment by Emily-Jiang** [comment]
> > > > > Looks good. A couple of notes:
> > > > > 
> > > > > * Are we using this for a final release of Reactive Messaging 2.0? If so, I believe the WG would not accept the CI to be in the staging repo when validating the binaries (we still need to change this into a more manageable release process).
> > > > 
> > > > 
> > > > No, it's using an RC. Even if the RC is in Maven Central, I really don't want to have this release be in Maven Central. It's there to validate that we pass the TCK, not something to be used by the end-user. There are many more steps to do before that point. With dependabot, if we push this into Central, you can be sure that tomorrow, you get people updating.
> > > 
> > > 
> > > I agree. I don't like us requiring it as well, but let's see if we can do it this way.
> > 
> > 
> > This was discussed on [the mailinglist](https://groups.google.com/g/microprofile/c/6v65mBjIXiQ) and we settled on RCx reelase for CCR as long as the final release is same as RCx.
> > This CCR looks good. I will kick off the release process with this.
> 
> I'm not completely sure if everyone has the same understanding. I'm fine like this, but we may need to document this and put it through an approval to make sure all of use are on the same page.

I will write a wiki to document the CCR requirement and plan to refine it with any new feedback.

**Comment by kwsutter** [comment]
Question...  Why are there so many callstacks in the TCK output?  If they are expected Exceptions, I can't tell that they are expected.  If it was just a couple of them, I'd let it go.  But, there are so many callstacks in the output, it makes it very difficult to decipher what's good and what's not.
